docs/notebooks/feasible_sets.pct.py

```diff
@@ -0,0 +1,224 @@
+#
+# Note that if we are interested in a feasibility region instead, it is simply a complement of the failure region, space with the value below 80.
+#
+# We illustrate the thresholded Branin function below, you can note that above the threshold of 80 there are no more values observed.
+
+# %%
+from trieste.objectives import branin, BRANIN_SEARCH_SPACE
+from util.plotting_plotly import plot_function_plotly
+from trieste.space import Box
+
+search_space = BRANIN_SEARCH_SPACE
+
+# threshold is arbitrary, but has to be within the range of the function
+threshold = 80
+
+# define a modified branin function
+def thresholded_branin(x):
+    y = np.array(branin(x))
+    y[y > threshold] = np.nan
+    return tf.convert_to_tensor(y.reshape(-1, 1), x.dtype)
+
+# illustrate the thresholded branin function
+fig = plot_function_plotly(
+    thresholded_branin, search_space.lower, search_space.upper, grid_density=700
+)
+fig.update_layout(height=800, width=800)
+fig.show()
+
+
+# %% [markdown]
+# We start with a small initial dataset where our expensive-to-evaluate function is evaluated on points coming from a space-filling Halton sequence.
+
+# %%
+import trieste
+
+observer = trieste.objectives.utils.mk_observer(branin)
+
+num_initial_points = 12
+initial_query_points = search_space.sample_halton(num_initial_points)
+initial_data = observer(initial_query_points)
+
+
+# %% [markdown]
+# ## Probabilistic model of the objective function
+#
+# Just like in sequential optimization, we use a probabilistic model of the objective function. Acquisition functions will exploit the predictive posterior of the model to identify the failure region. We use a `GPR` model from the GPflow library to formulate a Gaussian process model, wrapping it in Trieste's `GaussianProcessRegression` model wrapper.
+
+# %%
+import gpflow
+from trieste.models.gpflow.models import GaussianProcessRegression
+
+
+def build_model(data):
+    variance = tf.math.reduce_variance(data.observations)
+    kernel = gpflow.kernels.Matern32(variance=variance, lengthscales=[2, 2])
+    gpr = gpflow.models.GPR(data.astuple(), kernel, noise_variance=1e-5)
+    gpflow.set_trainable(gpr.likelihood, False)
+
+    return GaussianProcessRegression(gpr)
+
+
+model = build_model(initial_data)
+
+
+# %% [markdown]
+# ## Active learning with Expected feasibility acquisition function
+#
+# The problem of identifying a failure or feasibility region of a (expensive-to-evaluate) function $f$ can be formalized as estimating the excursion set, $\Gamma* = \{ x \in X: f(x) \ge T\}$, or estimating the contour line, $C* = \{ x \in X: f(x) = T\}$, for some threshold $T$ (see <cite data-cite="bect2012sequential"/> for more details). 
```

Comment:
links to citations do not work on the pdf export - just make sure they are OK!

Reply:
last item on the list, building the docs now